src/main/java/com/iridium/iridiumskyblock/listeners/PlayerRespawnListener.java

```diff
@@ -0,0 +1,18 @@
+package com.iridium.iridiumskyblock.listeners;
+
+import com.iridium.iridiumskyblock.IridiumSkyblock;
+import com.iridium.iridiumskyblock.database.User;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerRespawnEvent;
+
+public class PlayerRespawnListener implements Listener {
+
+    @EventHandler
+    public void onRespawn(PlayerRespawnEvent event) {
+        if (IridiumSkyblock.getInstance().getConfiguration().respawnOnIsland) {
+            User user = IridiumSkyblock.getInstance().getUserManager().getUser(event.getPlayer());
+            user.getIsland().ifPresent(island -> event.setRespawnLocation(island.getHome()));
```

Comment:
I think you should check if the current respawn location is already on the island because of beds, respawn anchors etc